python_jsonschema_objects/validators.py

```diff
@@ -95,13 +108,13 @@ def check_boolean_type(param, value, _):
 
 @type_registry.register(name='integer')
 def check_integer_type(param, value, _):
-    if not isinstance(value, int):
+    if not isinstance(value, int) or isinstance(value, bool):
```

Comment:
Those lines explicitly disallow assigning a boolean value to a property that has been declared to be of type "integer" or "number". They are necessary because python subclasses bool from int (PEP285) and will silently promote a bool to an int if we do not include an explicit check.

Sent from Outlook Mobile




On Fri, Feb 12, 2016 at 5:20 AM -0800, "Chris W" <notifications@github.com> wrote:












In python_jsonschema_objects/validators.py:
> @@ -95,13 +108,13 @@ def check_boolean_type(param, value, _):
>  
>  @type_registry.register(name='integer')
>  def check_integer_type(param, value, _):
> -    if not isinstance(value, int):
> +    if not isinstance(value, int) or isinstance(value, bool):



Is there a reason you changed this to allow booleans for integer and number values? 



—
Reply to this email directly or view it on GitHub.